src/screens/WalletDetailsScreen.tsx

```diff
@@ -82,35 +78,47 @@ export class WalletDetailsScreen extends React.PureComponent<Props, State> {
     });
   };
 
-  navigateWithWallet = (route: Route) =>
-    this.props.navigation.navigate(route as any, {
-      wallet: this.props.route.params.wallet,
+  navigateWithWallet = (route: Route.ExportWalletXpub | Route.ExportWallet) => {
+    const { navigation, wallet } = this.props;
+    if (!wallet) {
+      return;
+    }
+    navigation.navigate(route, {
+      wallet,
     });
+  };
 
   setLabel = (label: string) => {
     const trimmedlabel = label.trim();
-    const { wallet } = this.props.route.params;
-    this.props.navigation.setParams({ wallet });
-    this.setState({ label: trimmedlabel });
-    wallet.setLabel(trimmedlabel);
-    this.props.updateWallet(wallet);
-    BlueApp.saveToDisk();
+    const { wallet, updateWallet } = this.props;
+    if (!wallet) {
+      return;
+    }
+    const updatedWalelt = cloneDeep(wallet);
+    updatedWalelt.setLabel(trimmedlabel);
+    updateWallet(updatedWalelt);
   };
 
   editAmount = () => {
-    this.props.navigation.navigate(Route.EditText, {
+    const { wallet, navigation } = this.props;
+    if (!wallet) {
+      return;
+    }
```

Comment:
if there is no wallet in the props user will not see this screen according to the statement in the render method

Reply:
yes, but the TS is complaining without this condition, it is not that smart :D